src/main/zapHomeFiles/hud/components/AlertDetailsModal.vue

```diff
@@ -0,0 +1,120 @@
+            <td>{{ details['description'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_risk'"></td>
+            <td>{{ details['risk'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_confidence'"></td>
+            <td>{{ details['confidence'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_parameter'"></td>
+            <td>{{ details['parameter'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_attack'"></td>
+            <td>{{ details['attack'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_evidence'"></td>
+            <td>{{ details['evidence'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_cweid'"></td>
+            <td>{{ details['cweid'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_wascid'"></td>
+            <td>{{ details['wascid'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_other'"></td>
+            <td>{{ details['other'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_solution'"></td>
+            <td>{{ details['solution'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_reference'"></td>
+            <td v-if="details['reference']">
+              <li v-for="link in details['reference'].split('\n')">
+                <a :href="link" target="_top">{{ link }}</a>
+              </li>
+            </td>
+            <td v-else></td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+  </NavModal>
+</template>
+
+<script>
+import { EventBus } from "../libs/event-bus.js";
+import NavModal from "./NavModal.vue";
+
+export default {
+  template: "#alert-details-modal-template",
+  props: ["show", "title"],
+  components: {
+    NavModal
+  },
+  methods: {
+    close: function() {
+      this.$emit("close");
+    },
+    messageSelected: function(id) {
+      //TODO: replace tabId, frameId w/ props
+      navigator.serviceWorker.controller.postMessage({
+        tabId: tabId,
+        frameId: frameId,
+        action: "showHttpMessageDetails",
+        tool: "history",
+        id: id
+      });
+    },
+    back: function() {
+      //TODO: replace app (Vue context w/ props)
+      app.keepShowing = true;
+      app.isAlertDetailsModalShown = false;
+      this.port.postMessage({ back: true });
+    }
+  },
+  data() {
+    return {
+      port: null,
+      details: {}
+    };
+  },
+  created() {
+    let self = this;
+
+    EventBus.$on("showAlertDetailsModal", data => {
+      //TODO: replace app (Vue context w/ props)
+      app.isAlertDetailsModalShown = true;
+      app.alertDetailsModalTitle = data.title;
+
+      self.details = data.details;
+      self.port = data.port;
+    });
+  }
+};
+</script>
+
+<style lang="scss" scoped>
+</style>
```

Comment:
Can you add a new-line at eof?

Reply:
@jsoref Can you clarify the need or value add of adding a new-line? I'm just curious if that's compliance w/ a tool or some other purpose.

Reply:
git diff (and every other diff) get really pissy about this.

Basically all tools that work on whole lines (newline terminated) make optimizations based on this. So, whenever it's violated, they have to do a lot of work to undo this assumption.

And standard text editors (e.g. `vi`) will by default insert the newline. This means that if I (or someone else) use such an editor to make a change to a file which is missing the newline, the editor will add it, and `git` (or `hg` or ...) will blame us for changing that last line -- even if our change is in fact nowhere near. Once our change is merged, anyone who uses a blame tool will see us as having made the change to that last line, and if further lines are added below, it'll look like we changed that specific line.

So, as a general rule, the best policy is to always include the newline.

Reply:
Gotcha, thanks for the clarification. Seems like something a linter should be responsible for enforcing ultimately.